src/java/org/apache/cassandra/metrics/AccordStateCacheMetrics.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.metrics;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import com.codahale.metrics.Histogram;
+import org.apache.cassandra.cache.CacheSize;
+
+import static org.apache.cassandra.metrics.CassandraMetricsRegistry.Metrics;
+
+public class AccordStateCacheMetrics extends CacheMetrics
+{
+    public final Histogram objectSize;
```

Comment:
For the global case, is this `objectSize` metric different than the exposed `weightedSize()`? I overlooked this in my earlier reading...

Reply:
While `weightedSize` is a gauge on the number of bytes stored in the cache, `objectSize` is a histogram filled with size updates - that is, it is focused on size of the individual objects added or removed from the cache. The only redundant thing would be `objectSize.getCount()` which should be equal to `weightedSize`. 

I was actually going to ask whether it should be updated only with the sizes of objects added to the cache, ignoring what was dropped?

Reply:
Okay, that makes sense. In that case, yes, I would think that we'd only want to update the metric on add, as removal should be symmetric?

(To be honest, I have no idea what the distribution of sizes is going to be here. If sizes are all in a pretty narrow range, I'm not sure what the metric buys an operator.)

Reply:
This is how I understood the requested "Size of CommandsForKey when loaded" mentioned in the Jira ticket;

"should be symmetric" - eventually - yes

I'll update it to report only on addition to the cache

Reply:
actually, I was wrong, it is already implemented to report only additions, so nothing to do here